cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -122,6 +129,32 @@ public static DataprocClient fromConf(DataprocConf conf) throws IOException, Gen
     String subnet = conf.getSubnet();
     Network networkInfo = getNetworkInfo(projectId, network, compute);
 
+    boolean useInternalIP = !conf.isPreferExternalIP();
```

Comment:
this isn't correct. use internal IP should only be true if:

prefer external IP is false
AND
(system network is the same as the dataproc network) OR (system network is peered with dataproc network)

Reply:
updated code as discussed.